pkg/collector/corechecks/systemd/systemd.go

```diff
@@ -196,31 +199,33 @@ func (c *SystemdCheck) Run() error {
 
 func (c *SystemdCheck) connect(sender aggregator.Sender) (*dbus.Conn, error) {
 	conn, err := c.getDbusConnection()
-
 	if err != nil {
 		newErr := fmt.Errorf("cannot create a connection: %v", err)
 		sender.ServiceCheck(canConnectServiceCheck, metrics.ServiceCheckCritical, "", nil, newErr.Error())
 		return nil, newErr
 	}
+	sender.ServiceCheck(canConnectServiceCheck, metrics.ServiceCheckOK, "", nil, "")
+	return conn, nil
+}
 
+func (c *SystemdCheck) submitSystemdState(sender aggregator.Sender, conn *dbus.Conn) {
 	systemStateProp, err := c.stats.SystemState(conn)
+	// Expected to fail for Systemd version <212
+	// Source: https://github.com/systemd/systemd/blob/d4ffda38716d33dbc17faaa12034ccb77d0ed68b/NEWS#L7292-L7300
+	// TODO: Instead of logging debug, replace with condition based on Systemd version.
 	if err != nil {
-		newErr := fmt.Errorf("err calling SystemState: %v", err)
-		sender.ServiceCheck(canConnectServiceCheck, metrics.ServiceCheckCritical, "", nil, newErr.Error())
-		return nil, newErr
-	}
-	sender.ServiceCheck(canConnectServiceCheck, metrics.ServiceCheckOK, "", nil, "")
-
-	serviceCheckStatus := metrics.ServiceCheckUnknown
-	systemState, ok := systemStateProp.Value.Value().(string)
-	if ok {
-		status, ok := systemdStatusMapping[systemState]
+		log.Debugf("err calling SystemState: %v", err)
+	} else {
+		serviceCheckStatus := metrics.ServiceCheckUnknown
+		systemState, ok := systemStateProp.Value.Value().(string)
```

Comment:
```suggestion
		systemState, ok := systemStateProp.Value.String()
```
Just curious if that wouldn't work better

Reply:
The `Variant.String()` method is quite complex and will also return a string with quotes around e.g. `"running"`: https://github.com/godbus/dbus/blob/06fc4b473149e499166adbb9e31c7365a8ea146f/variant.go#L51-L139

Reply:
👍